control/trajectory_follower/design/trajectory_follower-design.md

```diff
@@ -12,9 +12,43 @@ This package provides the library code used by the nodes of the `trajectory_foll
 Mainly, it implements two algorithms:
 
 - Model-Predictive Control (MPC) for the computation of lateral steering commands.
-  - [trajectory_follower-mpc-design](trajectory_follower-mpc-design.md)
+  - [trajectory_follower-mpc-design](mpc_lateral_controller-design.md)
 - PID control for the computation of velocity and acceleration commands.
-  - [trajectory_follower-pid-design](trajectory_follower-pid-design.md)
+  - [trajectory_follower-pid-design](pid_longitudinal_controller-design.md)
+
+## Design
+
+![Controller](../../trajectory_follower_nodes/design/media/Controller.drawio.svg)
+
+There are lateral and longitudinal base interface classes and each algorithm inherits from this class to implement.
+The interface class has the following base functions.
+
+- `setInputData()`: Input the data subscribed in [controller_node](../../trajectory_follower_nodes/design/trajectory_follower-design.md). This must be implemented with the inherited algorithm and the used data must be selected.
+- `run()`: Compute control commands and return to [controller_node](../../trajectory_follower_nodes/design/). This must be implemented by inherited algorithms.
+- `syncData()`: Input the result of running the other controller.
```

Comment:
Would you mention the case of waiting steering convergence as an example?

Reply:
added the example https://github.com/autowarefoundation/autoware.universe/pull/1591/commits/5d4474950098ed4712bb7745d95add1273b39caa